### PR TITLE
Enabled more linters.

### PR DESCRIPTION
**Pull request by alanconway**
Enabled most of the available linters in golangci_lint.
Made minor code fixes to conform to the linters. No functional changes.

This PR has multiple commits to show what changes were for what linters.
If all are approved I will squash before merge.

/hold
/cc @periklis
/cc @jcantrill

### CODE REVIEW

File: golangci.yaml [hunk 1]
```diff
@@ -1,12 +1,35 @@
...
+  disable-all: true
   enable:
+    - bodyclose
+    - deadcode
+    - depguard
+    - dogsled
+    - errcheck
+    - gocyclo
+    - gofmt
+    - goimports
```
Comment (periklis) [inline golangci.yaml:11]: `goimports` assumes that `make fmt` is using `goimports`.

File: golangci.yaml [hunk 2]
```diff
@@ -1,12 +1,35 @@
...
+    - goprintffuncname
+    - gosec
     - gosimple
+    - govet
+    - ineffassign
+    - interfacer
```
Comment (periklis) [inline golangci.yaml:17]: `interfacer ` is deprecated from its maintainer being nasty making bad suggestions.

File: golangci.yaml [hunk 3]
```diff
@@ -1,12 +1,35 @@
...
+    - nakedret
+    - nolintlint
+    - rowserrcheck
```
Comment (periklis) [inline golangci.yaml:20]: `rowserrcheck` is not applicable in this repo, as we don't make use of golang's `sql` package.

File: golangci.yaml [hunk 4]
```diff
@@ -1,12 +1,35 @@
...
+    - scopelint
```
Comment (periklis) [inline golangci.yaml:21]: `scopelint` is superseded by `exportloopref`.

File: golangci.yaml [hunk 5]
```diff
@@ -1,12 +1,35 @@
...
     - staticcheck
+    - structcheck
+    - typecheck
+    - unconvert
+    - varcheck
+    - gocritic
```
Comment (periklis) [inline golangci.yaml:27]: `gocritic ` is highly opinionated, but let's see if we get used to it.

File: golangci.yaml [hunk 6]
```diff
@@ -1,12 +1,35 @@
...
     - unused
-    - interfacer
-    - errcheck
-    - prealloc
-    - gosec
+    - misspell
+    - dupl
```
Comment (periklis) [inline golangci.yaml:30]: `dupl` is nice but may be a source of headaches when using `client-go` structs or api-machinery's `metav1` package.

File: golangci.yaml [hunk 7]
```diff
@@ -1,12 +1,35 @@
...
 
+# Consider enabling these, some cleanup required:
+#    - stylecheck  # Replacement for golint, valid but numerous.
+#    - gomnd       # Avoid magic numbers.
+#    - whitespace  # Consistent use of whitespace.
```
Comment (periklis) [inline golangci.yaml:35]: I suggest to add `wsl` to the list above instead of `whitespace` only.

**Review by periklis — CHANGES_REQUESTED** [review]
In general LGTM, only a couple of points to consider:
- Linting for `goimports` should also reflect that we use `goimports` in `make fmt`
- A couple of suggested linters are obsolete because of harmful suggestions or superseded by others.
- I suggest to add `wsl` to the standard repertoire for increased code readbility.

File: golangci.yaml (hunk 2 above)
Comment (alanconway) [reply golangci.yaml:17]: +1

File: golangci.yaml (hunk 3 above)
Comment (alanconway) [reply golangci.yaml:20]: +1

File: golangci.yaml (hunk 4 above)
Comment (alanconway) [reply golangci.yaml:21]: +1

File: golangci.yaml (hunk 5 above)
Comment (alanconway) [reply golangci.yaml:27]: That one was interesting - I was fully expecting to eliminate it for noise, but actually out our code was already almost 100% compliant! It seems like we share its opinions :) We can remove if it turns out to be a pain.

File: golangci.yaml (hunk 4 above)
Comment (alanconway) [reply golangci.yaml:21]: exportloopref isn't supported by golangci-lint 1.27 and I am having trouble upgrading it. Can we put this in with scoprelint and upgrade/replace in a subsequent merge?

File: golangci.yaml (hunk 7 above)
Comment (alanconway) [reply golangci.yaml:35]: +1

File: golangci.yaml (hunk 6 above)
Comment (alanconway) [reply golangci.yaml]: Moved it to the wish list as it is reporting some very valid cut/paste abuse in the tests and I'd rather not take on any more in this merge. Would like to enable later.

File: golangci.yaml (hunk 1 above)
Comment (alanconway) [reply golangci.yaml:11]: I'm having trouble installing goimports with bingo. I've put it on the commented wish list for now. I'd like to come back to it but I want to get this merge in ASAP as it touches a lot of files and will become a merge-out nightmare quickly.

**Comment by alanconway** [comment]
@periklis agreed on all points. Have addressed some and left others for a future merge. This merge touches a lot of files and will get out of hand quickly if it stays out too long. LGTM?

File: golangci.yaml (hunk 4 above)
Comment (periklis) [reply golangci.yaml:18]: I am ok with doing this later.

**Review by periklis — APPROVED** [review]
LGTM

**Comment by alanconway** [comment]
> LGTM

I'll be needing that in /lgtm form if you please :)

/unhold

**Review by periklis — APPROVED** [review]
/lgtm

**Pull request merged**